### PR TITLE
feat(alignMinDiff): extend to any nb of components

### PR DESCRIPTION
**Pull request by opatiny**
Closes: #402

### CODE REVIEW

**Comment by opatiny** [comment]
The PR is failing because of a bug in `ml-matrix`, see https://github.com/mljs/matrix/issues/175

**Comment by targos** [comment]
You can rebase on `main` to pick up https://github.com/image-js/image-js-typescript/pull/404

File: src/filters/autoLevel.ts [hunk 1]
```diff
@@ -16,11 +21,29 @@ export interface AutoLevelOptions {
...
+  let min: number | number[] = minMax.min;
+  let max: number | number[] = minMax.max;
+
+  if (uniform) {
+    let minDiffIndex = -1;
+    let previousDiff = -1;
+    for (let i = 0; i < minMax.max.length; i++) {
+      const difference = minMax.max[i] - minMax.min[i];
+      if (difference > previousDiff) {
+        minDiffIndex = i;
```
Comment (targos) [inline src/filters/autoLevel.ts:39]: Isn't it a max diff?

Comment (opatiny) [reply src/filters/autoLevel.ts]: Yes, my bad '^^

**Review by targos — APPROVED** [review]

**Pull request merged**